src/cursor/session.rs

```diff
@@ -235,23 +340,17 @@ where
     }
 }
 
-impl<T> SessionCursor<T>
-where
-    T: DeserializeOwned + Unpin,
-{
+impl<T> SessionCursor<T> {
     fn mark_exhausted(&mut self) {
-        self.info.id = 0;
+        self.state.as_mut().unwrap().exhausted = true;
```

Comment:
now that state is the single source of truth, this needed to be updated to use `exhausted` instead of the spec.